File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
import java.util.Arrays;

class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        
        for(int i = 0; i < n; i++) {
            String s = "";
            int num = arr1[i] | arr2[i];
            s += Integer.toBinaryString(num);
            
            while(s.length() < n) {
                s = "0" + s;
            }
            
            s = s.replace("1", "#");
            s = s.replace("0", " ");
            answer[i] = s;
        }
    
        
        return answer;
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []

    for i in range(0, n):
        curr = arr1[i] | arr2[i]
        binary = bin(curr)[2:].rjust(n, '0')
        trans = str.maketrans({'1' : '#', '0' : ' '})
        answer.append(binary.translate(trans))

    return answer